src/iuse_actor.cpp

```diff
@@ -707,11 +707,9 @@ long pick_lock_actor::use( player *p, item *it, bool, const tripoint& ) const
             g->add_event( EVENT_WANTED, int( calendar::turn ) + 300, 0, p->global_sm_location() );
         }
     }
-    // Special handling, normally the item isn't used up, but it is if broken.
     if( it->damage >= 5 ) {
-        return 1;
-    }
-
+        p->i_rem(it);
```

Comment:
And after the item is removed (and `it` is invalid), the function returns `it->type->charges_to_use()`, which accesses `it` again. There should probably be a `return 0` after `i_rem` or so.